dolphinscheduler-task-plugin/dolphinscheduler-task-jupyter/src/main/java/org/apache/dolphinscheduler/plugin/task/jupyter/JupyterTask.java

```diff
@@ -0,0 +1,192 @@
+        super(taskExecutionContext);
+        this.taskExecutionContext = taskExecutionContext;
+    }
+
+    @Override
+    public void init() {
+
+        logger.info("jupyter task params {}", taskExecutionContext.getTaskParams());
+
+        jupyterParameters = JSONUtils.parseObject(taskExecutionContext.getTaskParams(), JupyterParameters.class);
+
+        if (null == jupyterParameters) {
+            logger.error("jupyter params is null");
+            return;
+        }
+
+        if (!jupyterParameters.checkParameters()) {
+            throw new RuntimeException("jupyter task params is not valid");
+        }
+    }
+
+    /**
+     * create command
+     *
+     * @return command
+     */
+    @Override
+    protected String buildCommand() {
+        /**
+         * papermill [OPTIONS] NOTEBOOK_PATH [OUTPUT_PATH]
+         */
+        List<String> args = new ArrayList<>();
+
+        args.add(JupyterConstants.CONDA_ACTIVATE);
+        args.add(jupyterParameters.getCondaEnvName());
+        args.add(JupyterConstants.JOINTER);
+        args.add(JupyterConstants.PAPERMILL);
+        args.add(jupyterParameters.getInputNotePath());
+        args.add(jupyterParameters.getOutputNotePath());
+
+        // populate jupyter parameterization
+        args.addAll(populateJupyterParameterization());
+
+        // populate jupyter options
+        args.addAll(populateJupyterOptions());
+
+        // replace placeholder, and combining local and global parameters
+        Map<String, Property> paramsMap = ParamUtils.convert(taskExecutionContext, getParameters());
+        if (MapUtils.isEmpty(paramsMap)) {
+            paramsMap = new HashMap<>();
+        }
+        if (MapUtils.isNotEmpty(taskExecutionContext.getParamsMap())) {
+            paramsMap.putAll(taskExecutionContext.getParamsMap());
+        }
+
+        String command = ParameterUtils.convertParameterPlaceholders(String.join(" ", args), ParamUtils.convert(paramsMap));
+
+        logger.info("jupyter task command: {}", command);
+
+        return command;
+    }
+
+
+    /**
+     * build jupyter parameterization
+     *
+     * @return argument list
+     */
+    private List<String> populateJupyterParameterization() {
+        List<String> args = new ArrayList<>();
+        String parameters = jupyterParameters.getParameters();
+        if (StringUtils.isNotEmpty(parameters)) {
+            ObjectMapper mapper = new ObjectMapper();
+            try {
+
+                // convert JSON string to Map
+                Map<String, String> jupyterParamsMap = mapper.readValue(parameters, Map.class);
+                for (String key : jupyterParamsMap.keySet()) {
+                    args.add(JupyterConstants.PARAMETERS);
+                    args.add(key);
+                    args.add(jupyterParamsMap.get(key));
+                }
+
+            } catch (IOException e) {
+                logger.error("fail to parse jupyter parameterization", e);
```

Comment:
Should it throw exception here?

Reply:
Thanks for pointing this out, I think I need to find a way to handle this exception. If I throw it here, do you have any suggestions where and how should I deal with this exception?

Reply:
I think if the jupyter parameterization is necessary, it should throw exception and make task fail.

Reply:
> I think if the jupyter parameterization is necessary, it should throw exception and make task fail.

I agree. Will fix it in the next commit.

Reply:
Fixed in the latest commit.